File: C/Program11.c
/////////////////////////////////////////////////////////////////////////
//                                                                     //
//  File name :     Program11.c                                        //
//  Description :   Used to Display Factorial of given number          //
//  Author :        Pallavi Omprakash Malewar                          //
//  Date :          28/04/2025                                         //
//                                                                     //
/////////////////////////////////////////////////////////////////////////

#include<stdio.h>

#define ERR_INVALID -1

unsigned long Factorial(unsigned int iNo)
{
    int iCount = 0;

    unsigned long iFactorial = 1;

    if(iNo < 0)
    {
        return ERR_INVALID;
    }
    
    iCount = 1 ;
    while(iCount <= iNo)
    {
        iFactorial = iFactorial * iCount;
        iCount++;
    }

    return iFactorial;

}


int main()
{
    unsigned int iValue = 0;
    unsigned long iRet = 0;

    printf("Enter Number : ");
    scanf("%u" , &iValue);

    iRet = Factorial(iValue);

    if(iRet == ERR_INVALID)
    {
        printf("Invalid Input\n");
    }

    else
    {
        printf("Factorial is : %lu\n" , iRet);
    }

    return 0;
}

File: JAVA/Program11.java
//////////////////////////////////////////////////////////////////////////////////////////
//                                                                                      //
//  File name :     Program11.java                                                      //
//  Description :   Used to Display Sum of Factors and NonFactors of Given Number       //
//  Author :        Pallavi Omprakash Malewar                                           //
//  Date :          25/08/2025                                                          //
//                                                                                      //
//////////////////////////////////////////////////////////////////////////////////////////

import java.util.Scanner;

class Number
{
    public void sum(int iNo)
    {
        int iCount = 0, iSumFact = 0, iSumNonFact = 0;

        for(iCount = 1 ; iCount <= iNo ; iCount++)
        {
            if((iNo % iCount) == 0)
            {
                iSumFact += iCount;
            }
            else
            {
                iSumNonFact += iCount;
            }

        }
        System.out.println("Summation of Factors : " + iSumFact);
        System.out.println("Summation of Non-Factors : " + iSumNonFact);

    }
}

class Program11
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0, iRet = 0;

        System.out.print("Enter Number : ");
        iValue = sobj.nextInt();

        Number nobj = new Number();

        nobj.sum(iValue);


        sobj.close();
    }
}